src/main/java/com/google/firebase/remoteconfig/Template.java

```diff
@@ -46,11 +46,20 @@
 
   /**
    * Creates a new {@link Template}.
+   *
+   * @param etag The ETag of this template.
    */
-  public Template() {
-    parameters = new HashMap<>();
-    conditions = new ArrayList<>();
-    parameterGroups = new HashMap<>();
+  public Template(String etag) {
+    this.parameters = new HashMap<>();
+    this.conditions = new ArrayList<>();
+    this.parameterGroups = new HashMap<>();
+    this.etag = etag;
+  }
+
+  Template() {
+    this.parameters = new HashMap<>();
```

Comment:
`this(null)`

Reply:
I have to cast null to String here for this to work... `this((String)null)`. Let me know if this is okay to use.